### PR TITLE
Change sysctl_set var to be dynamic depending on where it is executed

### PR DESCRIPTION
**Pull request by manderson-hpe**
## Summary and Scope

_Summarize what has changed. Explain why this PR is necessary. What is impacted? Is this a new feature, critical bug fix, etc?_

_Is this change backwards incompatible, backwards compatible, or a backwards compatible bugfix?_

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [issue id](issue link)
* Change will also be needed in `<insert branch name here>`
* Future work required by [issue id](issue link)
* Documentation changes required in [issue id](issue link)
* Merge with/before/after `<insert PR URL here>`

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `<development system>`
  * Local development environment
  * Virtual Shasta

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by jsl-hpe — COMMENTED** [review]
I believe this is an OK use of the variable as we want the role to execute in all contexts. We should ensure that similar changes are made to other top level ansible plays that are included or used to configure NCN nodes (not just master nodes).

**Review by spillerc-hpe — COMMENTED** [review]
This also needs to be set on worker and storage nodes so the same change would be required in `ncn-worker_nodes.yml` and `ncn-storage_nodes.yml`

I have verified that this change results in the desired behaviour on surtur by running iuf `prepare-images` with this change in place and performing NCN personalization of ncn-w004.

**Comment by manderson-hpe** [comment]
> This also needs to be set on worker and storage nodes so the same change would be required in `ncn-worker_nodes.yml` and `ncn-storage_nodes.yml`
> 
> I have verified that this change results in the desired behaviour on surtur by running iuf `prepare-images` with this change in place and performing NCN personalization of ncn-w004.

Thanks for looking at it and testing it, I actually went with a different route and posted the new code. I discussed the ticket with others and made the decision to make the code a bit more efficient, although more verbose.

File: ansible/ncn_nodes.yml [hunk 1]
```diff
@@ -26,10 +26,22 @@
...
+        name: '{{ inventory_hostname }}'
+        groups: cfs_image
+      when: cray_cfs_image
+
+# Does not include images
 - hosts:
-   - Management_Master
-   - Management_Worker
-   - Management_Storage
+   - Management_Master:!cfs_image
```
Comment (jsl-hpe) [inline ansible/ncn_nodes.yml:42]: No need to bifurcate the playbook into two for this. We can contextually pass in sysctl set values exactly once in the playbook's vars section via:

      vars:
        sysctl_set: {{ not cray_cfs_image }}

**Pull request closed without merging**